napari/_qt/_tests/test_async_slicing.py

```diff
@@ -98,18 +120,72 @@ def test_async_slice_multiscale_image_on_zoom(qtbot, make_napari_viewer, rng):
     image.corner_pixels = np.array([[0, 2, 3], [0, 6, 7]])
     image.events.reload(Event('reload', layer=image))
 
-    wait_until_vispy_image_data_equal(qtbot, vispy_image, data[0][2, 2:6, 3:7])
+    wait_until_vispy_image_data_equal(qtbot, vispy_layer, data[0][2, 2:6, 3:7])
+
+
+def test_slicing_in_progress(make_napari_viewer, qtbot, rng):
+    viewer = make_napari_viewer()
+    data = rng.random((3, 4, 5))
+    layer = LockableImage(data)
+    vispy_layer = setup_viewer_for_async_slice(viewer, layer)
+    assert viewer.dims.current_step != (2, 0, 0)
+
+    layer = vispy_layer.layer
+    assert not viewer.slicing_in_progress
+    with layer.lock:
```

Comment:
From a quick look, I don't think this lock is doing anything because it only seems to be held here. Did I miss something?

For images, I think you could reuse `LockableData` from `test_layer_slicer` and then hold that so that `__getitem__` is blocked. Points are a bit harder since they coerce the input `data` to a numpy array, but you can set `Points._data` directly as in `test_submit_with_one_3d_points`

Reply:
Looks like you are right, I'll dig into this.

Reply:
There is something very fishy going on with the data handling in async points. I haven't sorted it out yet, I need to understand more about dimensions (data indices vs world dims vs slice indices... )

Reply:
Found a bug, @andy-sweet opened a PR: https://github.com/napari/napari/pull/5783